modules/ROOT/pages/server_release_notes.adoc

```diff
@@ -38,6 +39,123 @@
+
+=== Remove empty directories from the files_versions
+
+Empty directories were left when the contained versions were deleted or moved. Large installations might end up with too many of these empty directories. Now, when a version is deleted, the containing directory will also be deleted if there aren't any more versions inside. https://github.com/owncloud/core/pull/40499[#40499]
+
+=== Use correct themed l10n app folder when app lives outside of server root
+
+When an app_path is pointing outside of the ownCloud server root or uses an symlink under certain conditions the l10n folder points to an invalid location and results in a crash of the server. This happened due to the assumption that app paths always start with the server root path. https://github.com/owncloud/core/pull/40607[#40607]
+
+=== Fix share-into-share move scenario
+
+Public links were lost upon moving share into another share as the share owner was not correctly set. This has been now partially fixed. https://github.com/owncloud/core/pull/40612[#40612]
+
+=== Prevent creation of empty files/folders when no available quota
+
+Until now it was possible for users having 0 quota or who already reached the limit of their assigned quota to still create empty files/folders, which generated confusion. The PR fixes this behaviour. https://github.com/owncloud/core/pull/40567[#40567]
+
+=== Drag & Drop folders into public file upload
+
+Previously only files were accepted via drag & drop. Users can now also drag folders into a public link that has the filedrop flag. When adding a folder, the hierarchy is flattened out and all files are added without any subfolders. Name collisons are avoided as usual. https://github.com/owncloud/core/pull/40643[#40643]
+
+=== Notable Changes
+
+* The suggested host name and port syntax for the database host on the installation has been corrected. https://github.com/owncloud/core/pull/40348[#40348]
+* Without this change, server-side Apache Auth (e.g. Shibboleth) resulted in redirect to default owncloud page, instead of last visited page. We now correctly handle redirect_url. https://github.com/owncloud/core/pull/40470[#40470] https://github.com/owncloud/core/pull/40161[#40161]
+* Before this change, objectstorages were reporting only infinite storage space. This could have caused problems in other apps that rely on this storage method, e.g. metrics app that monitors available space. https://github.com/owncloud/core/pull/40192[#40192]
+* We now expect a not found error instead of permission denied error for some trash interactions. https://github.com/owncloud/core/pull/40406[#40406]
```

Comment:
```suggestion
* We now expect a not found error instead of a permission denied error for some trash interactions. https://github.com/owncloud/core/pull/40406[#40406]
```